packages/vm/src/evm/opcodes/functions.ts

```diff
@@ -1023,33 +1024,56 @@ export const handlers: Map<number, OpHandler> = new Map([
   [
     0xf6,
     async function (runState) {
-      const [commitUnpadded, yParity, r, s] = runState.stack.popN(4)
-      if (s > SECP256K1_ORDER_DIV_2) {
+      // eslint-disable-next-line prefer-const
+      let [authority, memOffset, memLength] = runState.stack.popN(3)
+
+      if (memLength > 128n) {
+        memLength = 128n
+      }
+
+      let mem = runState.memory.read(Number(memOffset), Number(memLength))
+      if (mem.length < 128) {
+        mem = setLengthRight(mem, 128)
+      }
```

Comment:
"If `length` is greater than 128, the extra bytes are ignored. Bytes outside the range (in the event `length` is less than 128) are treated as if they had been zeroes."

👍